### PR TITLE
ci(lint): enable errorlint linter

### PR DESCRIPTION
**Pull request by mmorel-35**
## What does this PR do?

Enables [errolint](https://golangci-lint.run/usage/linters/#errorlint) linter

## Why is it important?

Errorlint is a linter for that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13.

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -794,7 +794,8 @@ func (p *DockerProvider) BuildImage(ctx context.Context, img ImageBuildInfo) (st
 	err = backoff.Retry(func() error {
 		resp, err = p.client.ImageBuild(ctx, buildContext, buildOptions)
 		if err != nil {
-			if _, ok := err.(errdefs.ErrNotFound); ok {
+			var enf errdefs.ErrNotFound
+			if ok := errors.As(err, &enf); ok {
```
Comment (mdelapenya) [inline docker.go:798]: One question regarding style: should we stand with this ok pattern, or just use this one?
```suggestion
			if errors.As(err, &enf) {
```

The same would apply to all the changes below.

File: docker.go [hunk 2]
```diff
@@ -1161,7 +1162,8 @@ func (p *DockerProvider) attemptToPullImage(ctx context.Context, tag string, pul
 	err = backoff.Retry(func() error {
 		pull, err = p.client.ImagePull(ctx, tag, pullOpt)
 		if err != nil {
-			if _, ok := err.(errdefs.ErrNotFound); ok {
+			var enf errdefs.ErrNotFound
+			if ok := errors.As(err, &enf); ok {
```
Comment (mmorel-35) [inline docker.go:1166]:
```suggestion
			if errors.As(err, &enf) {
```

File: wait/host_port.go [hunk 3]
```diff
@@ -155,8 +155,10 @@ func externalCheck(ctx context.Context, ipAddress string, port nat.Port, target
 		}
 		conn, err := dialer.DialContext(ctx, proto, address)
 		if err != nil {
-			if v, ok := err.(*net.OpError); ok {
-				if v2, ok := (v.Err).(*os.SyscallError); ok {
+			var v *net.OpError
+			if ok := errors.As(err, &v); ok {
```
Comment (mmorel-35) [inline wait/host_port.go:159]:
```suggestion
			if errors.As(err, &v) {
```

File: wait/host_port.go [hunk 4]
```diff
@@ -155,8 +155,10 @@ func externalCheck(ctx context.Context, ipAddress string, port nat.Port, target
...
+				var v2 *os.SyscallError
+				if ok := errors.As(v.Err, &v2); ok {
```
Comment (mmorel-35) [inline wait/host_port.go:161]:
```suggestion
				if errors.As(v.Err, &v2) {
```

**Review by mdelapenya — APPROVED** [review]
LGTM!

**Pull request merged**